trieste/space.py

```diff
@@ -429,12 +796,42 @@ def __eq__(self, other: object) -> bool:
         if not isinstance(other, Box):
             return NotImplemented
         return bool(
-            tf.reduce_all(self.lower == other.lower) and tf.reduce_all(self.upper == other.upper)
+            tf.reduce_all(self.lower == other.lower)
+            and tf.reduce_all(self.upper == other.upper)
+            and self._constraints == other._constraints
```

Comment:
(maybe add a comment about how we're insisting on the exact same list of constraints?)

Reply:
Added in 30a17e7